fe/fe-core/src/main/java/com/starrocks/connector/ConnectorManager.java

```diff
@@ -0,0 +1,37 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+package com.starrocks.connector;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import javax.annotation.concurrent.GuardedBy;
+
+// ConnectorManager is responsible for managing all ConnectorFactory, and for creating Connector
+public class ConnectorManager {
+    @GuardedBy("this")
+    private final ConcurrentMap<String, ConnectorFactory> connectorFactories = new ConcurrentHashMap<>();
+
+    /**
+     * add connectorFactory
+     *
+     * @param connectorFactory - a connector factory instance
+     */
+    public synchronized void addConnectorFactory(ConnectorFactory connectorFactory) {
```

Comment:
`CouncurrentHashMap` could ensure thread safety. I think it's non-meaning using synchronized.